### PR TITLE
Enable xforms-product-type on Control Gallery for tests

### LINKED ISSUE #9436: [Bug] [iOS] --optimize=experimental-xforms-product-type additional mtouch argument causes app crash
### Description

Template app crashes when the --optimize=experimental-xforms-product-type additional mtouch argument is used along with the Link Framework SDKS linker option. 

This argument is used to avoid an Apple App Store warning about UIWebView. See:
https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/webview?tabs=windows#uiwebview-deprecation-and-app-store-rejection-itms-90809
and
https://github.com/xamarin/Xamarin.Forms/issues/7323

### Steps to Reproduce

1. Open Visual Studio 2019 or Visual Studio for Mac
2. Create a new template blank Xamarin.Forms app
3. Select Debug  configuration and an iPhone Simulator.
4. Set the iOS project as the Startup project.
5. Update the Xamarin.Forms to version 4.5.0.187-pre2 (which is the needed version to use the `--optimize=experimental-xforms-product-type` mtouch argument.
6. Open the iOS project properties -> iOS Build page
7. Add `--optimize=experimental-xforms-product-type` to the additional mtouch arguments field
8. Select "Link Framework SdKs Only" option for the Linker Behavior. 
9. Deploy the iOS app. 

### Expected Behavior
App launches and displays initial screen

### Actual Behavior
App crashes with following exception
```
2020-02-03 10:21:13.241644-1000 App1.iOS[61152:3467260] 
Unhandled Exception:
System.MissingMethodException: Default constructor not found for type Xamarin.Forms.Platform.iOS.LabelRenderer
```

### Basic Information

- Version with issue: 4.5.0.187-pre2
- Last known good version: Unknown
- IDE: VS 2019 or VS for Mac
- Platform Target Frameworks: 
  - iOS:  Xamarin.iOS 13.10.0.17 
- Nuget Packages: Xamarin.Essentials, Xamarin.Forms

### Workaround

The workaround is to not use this additional mtouch argument and tolerate the warning from the App Store. ~~However in April Apple will start rejecting apps that have the UIWebView in them rather than just a warning.~~

* Apple won't accept **new** application using `UIWebView` in April.
* Apple won't accept application **updates using** `UIWebView` in December.

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###
Enable xforms-product-type on Control Gallery 

### Issues Resolved ### 
- fixes #9436 

### Platforms Affected ### 
- iOS

### Testing Procedure ###
- follow the steps that are part of #9436 to see if you can recreate the issue and then if this nuget fixes the issue

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.iOS/CollectionView/ItemsViewLayout.cs [hunk 1]
```diff
@@ -214,8 +214,7 @@ protected void DetermineCellSize()
 				return;
 			}
 
-			// Constrain and measure the prototype cell
-			prototype.ConstrainTo(ConstrainedDimension);
+			prototype.ConstrainTo(ConstrainedDimension);
```
Comment (PureWeen) [inline Xamarin.Forms.Platform.iOS/CollectionView/ItemsViewLayout.cs]:
```suggestion
			// Constrain and measure the prototype cell
			prototype.ConstrainTo(ConstrainedDimension);			
```

File: Xamarin.Forms.Platform.iOS/CollectionView/ItemsViewLayout.cs [hunk 2]
```diff
@@ -215,7 +215,7 @@ protected void DetermineCellSize()
 			}
 
 			// Constrain and measure the prototype cell
-			prototype.ConstrainTo(ConstrainedDimension);
+			prototype.ConstrainTo(ConstrainedDimension);
```
Comment (PureWeen) [inline Xamarin.Forms.Platform.iOS/CollectionView/ItemsViewLayout.cs]:
```suggestion
			prototype.ConstrainTo(ConstrainedDimension);	
```

**Comment by kaminipatel03** [comment]
It is still not working for me. I have installed 4.5.0.266-pre3 xamarian.forms package.

**Comment by pavanalluri007** [comment]
it is not working for me. i get "Default constructor not found for type Xamarin.Forms.Platform.iOS.ActivityIndicatorRenderer" and app crashes

**Comment by kaminipatel03** [comment]
> it is not working for me. i get "Default constructor not found for type Xamarin.Forms.Platform.iOS.ActivityIndicatorRenderer" and app crashes

Actually after some investigation found that if set Deployment Target above version 10 then error will not come. but still issue of UIWebView email persist.

**Comment by pavanalluri007** [comment]
I have deployment target as 11.0 still when i add "--optimize=experimental-xforms-product-type" as Additional mtouch arguments we don't receive UIWebView email but app crashed with above error.

**Comment by kaminipatel03** [comment]
> I have deployment target as 11.0 still when i add "--optimize=experimental-xforms-product-type" as Additional mtouch arguments we don't receive UIWebView email but app crashed with above error.

Can you send me your what you set in Link Behaviour and Supported Architechure?
What is version of Xamarian.ios and Xamarian.forms?

**Comment by pavanalluri007** [comment]
> > I have deployment target as 11.0 still when i add "--optimize=experimental-xforms-product-type" as Additional mtouch arguments we don't receive UIWebView email but app crashed with above error.
> 
> Can you send me your what you set in Link Behaviour and Supported Architechure?
> What is version of Xamarian.ios and Xamarian.forms?

Link Behavior: Link Framework SDKs Only
Supported Architectures: ARM64
Xamarin.iOS: 13.10.0.17
Xamarin.Forms: 4.5.0.282-pre4

**Comment by kaminipatel03** [comment]
> > > I have deployment target as 11.0 still when i add "--optimize=experimental-xforms-product-type" as Additional mtouch arguments we don't receive UIWebView email but app crashed with above error.
> > 
> > 
> > Can you send me your what you set in Link Behaviour and Supported Architechure?
> > What is version of Xamarian.ios and Xamarian.forms?
> 
> Link Behavior: Link Framework SDKs Only
> Supported Architectures: ARM64
> Xamarin.iOS: 13.10.0.17
> Xamarin.Forms: 4.5.0.282-pre4

I have also same settings and it is work for me. But i have one problem with ARM64 is that it is allowed only  60 MB app size. this is not allowed me to upload on app store. feeling irritated now with getting everytime new issues.

**Comment by PureWeen** [comment]
This won't be available until 4.5 stable

If you want to test it out you can try the nightly builds

**Comment by PureWeen** [comment]
@kaminipatel03 @pavanalluri007 

If you could test the nightly
https://github.com/xamarin/Xamarin.Forms/wiki/Nightly-Builds

And see if your apps run that would be super helpful!!

**Pull request merged**